server/src/main/java/org/opensearch/common/cache/CacheUsage.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.cache;
```

Comment:
There is a fair bit of discussion going on related to refactoring/modularizing, and one of the problems is that we have java packages split between jars. In particular, `org.opensearch.common` is split between `server` and `libs/core`. To avoid making the problem worse, I'm inclined to move these new classes out of `org.opensearch.common`, maybe into `org.opensearch.index.store.remote` so that it is even more obvious they are internal and subject to change. What do you think?